FEAT: BDBD-333 글쓰기 화면 api 호출 테스트

Dorii0513 opened the pull request:
### 개요
conflict 해결을 못해서 최신화된 develop 브랜치로 다시 만들었습니다.
category, files, representPicture 필드는 아직 추가 못해서 계속 진행할 예정입니다 :)
### 변경사항
1. url은 `Constant` 파일 따로 생성해서 `Constant.domainURL`로 수정했습니다.
2. `WritingStruct`에서 `files`랑 `hopePrice`가 nullable한데, 아직 `files`는 구현을 못해서 일단 `hopePrice`만 필드 구분 해주었습니다.
### 관련 지라 및 위키 링크
- [BDBD-333](https://fake-developers.atlassian.net/jira/software/projects/BDBD/boards/10?selectedIssue=BDBD-333)
### 리뷰어에게 하고 싶은 말
열심히하겠습니다

Dorii0513 committed:
FEAT: BDBD-333 글쓰기 화면 api 호출 테스트

SungHyunC reviewed (approved):
깔끔하게 잘 하셨네요! 수고하셨습니다!

minseonglove commented on BidderBidder/Supporting File/Constant.swift:
```diff
@@ -0,0 +1,13 @@
+
+struct Constant: Codable{
+    static let domainURL = "http://bidderbidderapi.kro.kr:8080"
+}
```
포트가 리얼은 80 샌박은 8080이야
나중에 develop에 들어갈 때는 수정해줘야해

Dorii0513 replied:
넵 수정하겠습니다 :)

minseonglove commented on BidderBidder/Supporting File/WritingStruct.swift:
```diff
@@ -0,0 +1,29 @@
+    let openingBid: String?
+    let tick: String?
+    let expirationDate: String?
+    let productContent: String?
```
스위프트는 초기값이 없을 때 물음표를 붙이는구나

minseonglove commented on BidderBidder/Supporting File/WritingStruct.swift:
```diff
@@ -0,0 +1,29 @@
+    
+    
+    public init(productTitle: String?, hopePrice: String?, openingBid: String?, tick: String?, expirationDate: String?, productContent: String?){
+        self.productTitle = productTitle
```
그럼 매개변수는 물음표가 없어도 되지 않나

Dorii0513 replied:
음,,제가 알기론 매개변수에도 붙여줘야해요!

minseonglove replied:
> 음,,제가 알기론 매개변수에도 붙여줘야해요!

그렇구나

minseonglove commented on BidderBidder/Cell/ProductViewCell.swift:
```diff
@@ -40,7 +40,7 @@ class ProductViewCell: UITableViewCell {
     
     func setCell(product: Product) {
-        thumbnail.downloaded(from: Constant.domainURL+":"+Constant.sandbox+product.thumbnail)
+        thumbnail.downloaded(from: Constant.domainURL+"/product/getInfiniteProductList")
```
요부분은 롤백해줘 수정이 필요한건 맞지만 지금 작업범위가 아니야

Dorii0513 replied:
아 이부분이 자꾸 에러가 나서 수정했어요.. 
develop 브랜치에 Constant파일이 없어서 자꾸 에러가 생기더라구요,,,
그래서 그냥 Constant 파일 하나 만들어주고 안에 domainURL만 넣었어요!
Constant.sandbox 부분은 domainURL에 포함되게 넣어줘서 지우고 수정했습니다

minseonglove replied:
> 아 이부분이 자꾸 에러가 나서 수정했어요.. develop 브랜치에 Constant파일이 없어서 자꾸 에러가 생기더라구요,,, 그래서 그냥 Constant 파일 하나 만들어주고 안에 domainURL만 넣었어요! Constant.sandbox 부분은 domainURL에 포함되게 넣어줘서 지우고 수정했습니다

오류가 나서 수정을 하는건 괜찮은데 커밋 할 땐 작업 범위에 벗어나는 파일들은 제외하고 커밋을 해야해
안그러면 작업이 겹쳐져서 컨플릭트가 날 수 있어

Dorii0513 replied:
헉 그렇군요 ㅠ_ㅠ...
넵 다음부턴 주의하겠습니다!

minseonglove reviewed (changes requested):
힘들겠지만 files를 넣어서 요청하는건 이 작업에서 반드시 넘어야만 하는 산인 것이야
files까지 넣어서 요청에 성공하면 리뷰 재요청 해줘

Dorii0513 committed:
FIX: BDBD-333 url 주소 수정 및 오류 해결

Dorii0513 committed:
FIX: feature/BDBD-333 conflict 해결1

Dorii0513 committed:
FEAT: BDBD-333 files 추가

FIX: BDBD-333 코드스멜 files 변수명 수정

FIX: BDBD-333 에러 수정

FIX: BDBD-333 warning 수정

FIX: BDBD-333 사진선택 오류 해결

FIX: BDBD-333 gitignore 추가

FIX: BDBD-333 GoogleSevice-info.plist 삭제

FIX: BDBD-333 multiFormData로 코드 수정

FIX: BDBD-333 category field 추가

Dorii0513 commented:
리뷰 주신 부분 수정 하고 갤러리 사진 선택 및 저장, api 테스트까지 해보았습니다!

Dorii0513 committed:
FIX: BDBD-333 expirationDate 형식 수정

Dorii0513 committed:
FIX: UI화면 및 imagePicker 코드 수정

Dorii0513 committed:
FIX: UI화면 및 imagePicker 코드 수정

FIX: BDBD-333 api post 코드 수정

FIX:  BDBD-333 AF.upload 코드 .response로 수정

FIX : header 수정 및 api 코드 정비

FIX: validate() 삭제

FIX : BDBD-333 completionHandler 추가

FIX : BDBD-333 GoogleService-info.plist 파일 삭제

FIX : BDBD-333 GoggleService-info.plist local파일에 추가

Dorii0513 committed:
FIX: BDBD-333 api post 코드 수정

Dorii0513 committed:
FIX:  BDBD-333 AF.upload 코드 .response로 수정

Dorii0513 committed:
FIX : header 수정 및 api 코드 정비

Dorii0513 committed:
FIX: validate() 삭제

Dorii0513 committed:
FIX : BDBD-333 completionHandler 추가

konempty commented on BidderBidder/Supporting File/WritingPostService.swift:
```diff
@@ -0,0 +1,69 @@
+    
+    static let shared = WritingPostService()
+    
+    func postWritingData(productTitle: String,
```
이 함수는 UiViewController 클래스에 넣어주시고 파라메터도 잘 정리해서 sendRestRequest 메소드와 형식을 맞춰주세요!

konempty commented on BidderBidder/Supporting File/WritingPostService.swift:
```diff
@@ -0,0 +1,69 @@
+                         tick: Int,
+                         expirationDate: String,
+                         productContent: String,
+                         hopePrice: Int,
```
hopePrice는 값이 안들어 올수도 있고 값이 없으면 nil이 들어가야 해요!

konempty commented on BidderBidder/Supporting File/WritingPostService.swift:
```diff
@@ -0,0 +1,69 @@
+        AF.upload(multipartFormData: { (multipartFormData) in
+            
+            for image in files {
+                            if let image = image?.jpegData(compressionQuality: 1) {
```
이런 함수가 있었군요 좋습니다! 👍

konempty commented on BidderBidder/Supporting File/WritingStruct.swift:
```diff
@@ -0,0 +1,53 @@
+//        self.hopePrice = hopePrice
+//        self.files = files?.jpegData(compressionQuality: 1)
+//    }
+//}
```
불필요한 주석은 바로바로 지워주세요ㅎㅎ

konempty commented on BidderBidder/Supporting File/WritingPostService.swift:
```diff
@@ -0,0 +1,69 @@
+                return
+            }
+            print("success")
+            print(response)
```
디버깅 용도의 print문은 자제해주시면 좋을것 같아요!

konempty commented on BidderBidder/Controller/WritingViewController.swift:
```diff
@@ -0,0 +1,140 @@
+        let hopePrice = Int(hopePriceTextField.text!)
+        let openingBid = Int(openingBidTextField.text!)
+        let tick = Int(tickTextField.text!)
+        let expirationDate = Double(expirationDateTextField.text!)
```
자료형을 Double로 하신 이유가 있으실까요?

Dorii0513 replied:
엇.. Int로 받아도 될거 같은데 제가 뭔가 착각했었나봐요! 수정하겠습니다 :)

Dorii0513 replied:
아 다시보니 뒤에 `expirationDate`을 계산할 때, `timePlus()` 함수에서 사용하는 `TimeInterval` 메소드가 Double 타입이라서 자료형을 Double로 선언했었네요!

konempty commented on BidderBidder/Controller/WritingViewController.swift:
```diff
@@ -0,0 +1,140 @@
+        var config = YPImagePickerConfiguration()
+        
+        config.showsPhotoFilters = false
+        config.library.maxNumberOfItems = 3
```
3개로 제한하신 이유가 있으실까요?

Dorii0513 replied:
라이브러리에서 사진 선택을 무제한으로 할 수 있게 하는 기능이 따로 없어서, 테스트용으로 3으로 제한시키고 시도해봤어요!

konempty commented on BidderBidder/Controller/WritingViewController.swift:
```diff
@@ -0,0 +1,140 @@
+    @IBOutlet weak var productContentTextView: UITextView!
+    
+    //imageFiles
+    @IBOutlet var imageFiles: [UIImageView]!
```
엇 ㅠㅠ 사진의 개수는 정해지지 않았기 때문에 미리 배열을 생성할수 없을것 같아요

Dorii0513 replied:
엇 그러네요! 수정하겠습니다 :)

konempty commented on BidderBidder/Controller/WritingViewController.swift:
```diff
@@ -0,0 +1,140 @@
+            self.filesSelectButton.isHidden = true
+            picker.dismiss(animated: true)
+        }
+        present(picker, animated: true, completion: nil)
```
이 라이브러리가 정말 좋은것 같은데 사진을 선택하고 난뒤에 다시 선택하거나 사진을 뺄수 없고 사진의 순서를 정할 수 없는 문제만 해결되면 좋을것 같아요!

konempty reviewed (changes requested):
크흐 너무 고생하셨어요 ㅠㅠ 중간에 서버 문제인걸 빨리 알아 챘으면 조금 덜 고생 하셨을수 있었을것 같은데 
리뷰 달아드린거 수정 부탁 드리고 수정 다하시면 다시 리뷰요청 주세요!

konempty commented:
> Kudos, SonarCloud Quality Gate passed!    [![Quality Gate passed](https://camo.githubusercontent.com/e4a90cf346b46d44c5652111d68603714bfb7693043d07a95992de3a30bfce04/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f5175616c6974794761746542616467652f7061737365642d313670782e706e67)](https://sonarcloud.io/dashboard?id=FakeDevelopers_BidderBidder_IOS&pullRequest=12)
> 
> [![Bug](https://camo.githubusercontent.com/4c6102327f5a954f9c8acaf2e2714183157a9e41717b371b2cd585cf25057310/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f6275672d313670782e706e67)](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_IOS&pullRequest=12&resolved=false&types=BUG) [![A](https://camo.githubusercontent.com/1cba125a897d7fa47033a3b3b2be2bbee680d34d4f004a215564659b853fb201/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f526174696e6742616467652f412d313670782e706e67)](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_IOS&pullRequest=12&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_IOS&pullRequest=12&resolved=false&types=BUG) [![Vulnerability](https://camo.githubusercontent.com/3ba1ee49636ffc3427e38649a9f8a65ee392f28e8a662fcf96ce24cefbb520e9/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f76756c6e65726162696c6974792d313670782e706e67)](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_IOS&pullRequest=12&resolved=false&types=VULNERABILITY) [![A](https://camo.githubusercontent.com/1cba125a897d7fa47033a3b3b2be2bbee680d34d4f004a215564659b853fb201/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f526174696e6742616467652f412d313670782e706e67)](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_IOS&pullRequest=12&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_IOS&pullRequest=12&resolved=false&types=VULNERABILITY) [![Security Hotspot](https://camo.githubusercontent.com/fb735cbe76f8d5e1679c76ce83b740ceb1eaf62de4f7bf88623dc9953261aff7/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f73656375726974795f686f7473706f742d313670782e706e67)](https://sonarcloud.io/project/security_hotspots?id=FakeDevelopers_BidderBidder_IOS&pullRequest=12&resolved=false&types=SECURITY_HOTSPOT) [![A](https://camo.githubusercontent.com/1cba125a897d7fa47033a3b3b2be2bbee680d34d4f004a215564659b853fb201/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f526174696e6742616467652f412d313670782e706e67)](https://sonarcloud.io/project/security_hotspots?id=FakeDevelopers_BidderBidder_IOS&pullRequest=12&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=FakeDevelopers_BidderBidder_IOS&pullRequest=12&resolved=false&types=SECURITY_HOTSPOT) [![Code Smell](https://camo.githubusercontent.com/8fe18b2dfb6f7d4e44582f281b29f617eb5ae07c248d2002ca586e91da219212/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f636f64655f736d656c6c2d313670782e706e67)](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_IOS&pullRequest=12&resolved=false&types=CODE_SMELL) [![A](https://camo.githubusercontent.com/1cba125a897d7fa47033a3b3b2be2bbee680d34d4f004a215564659b853fb201/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f526174696e6742616467652f412d313670782e706e67)](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_IOS&pullRequest=12&resolved=false&types=CODE_SMELL) [8 Code Smells](https://sonarcloud.io/project/issues?id=FakeDevelopers_BidderBidder_IOS&pullRequest=12&resolved=false&types=CODE_SMELL)
> 
> [![No Coverage information](https://camo.githubusercontent.com/86785eaa572e061f5ecb3f8bf1cbb7b8eca44925083f92d067805b4362cead0c/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f436f76657261676543686172742f4e6f436f766572616765496e666f2d313670782e706e67)](https://sonarcloud.io/component_measures?id=FakeDevelopers_BidderBidder_IOS&pullRequest=12) No Coverage information [![0.0%](https://camo.githubusercontent.com/8047c63e1f9ed03f63001e1eadce4676bade3e0f83ec690a9c625287796248a6/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f4475706c69636174696f6e732f332d313670782e706e67)](https://sonarcloud.io/component_measures?id=FakeDevelopers_BidderBidder_IOS&pullRequest=12&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=FakeDevelopers_BidderBidder_IOS&pullRequest=12&metric=new_duplicated_lines_density&view=list)

@Dorii0513 코드스멜도 수정 부탁드릴게요!

minseonglove commented on BidderBidder/Supporting File/WritingPostService.swift:
```diff
@@ -0,0 +1,69 @@
+            for image in files {
+                            if let image = image?.jpegData(compressionQuality: 1) {
+                                multipartFormData.append(image, withName: "files", fileName: "\(files).jpg", mimeType: "image/jpeg") // jpeg 파일로 전송
+                                //print(image)
```
mimeType은 이미지에서 추출해서 사용하면 좋겠어
지금은 괜찮지만 나중에 gif나 webp도 지원할 때는 많이 곤란해져

minseonglove commented on BidderBidder/Controller/WritingViewController.swift:
```diff
@@ -0,0 +1,140 @@
+            
+            var i: Int = 0
+            
+            for item in items {
```
인덱스로 순회하면 바깥에 i가 없어도 되지 않을가

minseonglove reviewed (approved):
일이 잘 해결돼서 다행이야

Dorii0513 committed:
FIX : BDBD-333 GoogleService-info.plist 파일 삭제

Dorii0513 committed:
FIX : BDBD-333 postWritingData 함수 수정

Dorii0513 committed:
FIX: BDBD-333 hopePrice 옵셔널 적용

Dorii0513 committed:
FIX: BDBD-333 hopePrice parameter에 추가해서 다시 수정

Dorii0513 committed:
FIX: BDBD-333 Imagepicker index로 순환 하도록 수정

Dorii0513 committed:
FIX: BDBD-333 SwiftLint 적용

Dorii0513 committed:
FIX: config 파일 추가 및 URL 코드스멜 수정

Dorii0513 committed:
FIX: BDBD-333 fileName 수정
